ironfish/src/primitives/rawTransaction.ts

```diff
@@ -54,7 +55,20 @@ export class RawTransaction {
         throw new Error('Cannot post transaction. Mint value exceededs maximum')
       }
 
-      builder.mint(mint.asset, mint.value)
+      let asset: Asset
+      if (mint.assetId !== undefined) {
+        asset = new Asset(this.spendingKey, mint.name, mint.metadata)
+
+        if (!asset.id().equals(mint.assetId)) {
+          throw new ValidationError(
+            `Unauthorized to mint for asset '${mint.assetId.toString('hex')}'`,
+          )
+        }
```

Comment:
raw transaction shouldn't take an ID for this, since all this is already handled by the verifier.

Reply:
Nno, this logic is not handled by verifier. this is existing validation for mint authorization.

Reply:
Discussed with @rohanjadvani , he confirmed this existing validation is not necessary, can be skipped. I will remove it.